File: src/main/java/com/daddyrusher/bankdepositsapp/domain/Bank.java
package com.daddyrusher.bankdepositsapp.domain;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;

@Data
@Entity
@Table(name = "banks")
public class Bank {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    @NotBlank
    private String name;

    @Column(name = "bik_number")
    @NotBlank
    private Long bikNumber;
}

File: src/main/kotlin/com/daddyrusher/bankdeposits/domain/Bank.kt
package com.daddyrusher.bankdeposits.domain;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;

@Entity
@Table(name = "banks")
class Bank : AbstractEntity() {

    @Column(name = "name")
    @NotBlank
    private val name: String = ""

    @Column(name = "bik")
    @NotBlank
    private val bik: Long = 0
}
